src/coreclr/scripts/superpmi.py

```diff
@@ -517,19 +518,46 @@ def get_all_assemblies(location, root=True):
                     
                     return assemblies
 
+                def make_safe_filename(s):
```

Comment:
tempfile should be able to give you what you are trying to do here.

Reply:
@jashook I updated to use `mkstemp()`. Let me know if the new code is preferable.